rosidl_typesupport_connext_cpp/src/dds_utilities.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rosidl_typesupport_connext_cpp/dds_utilities.hpp>
+
+bool
+get_datareader_qos(DDSDomainParticipant * participant, DDS_DataReaderQos & datareader_qos)
```

Comment:
These functions are used in `srv__type_support.cpp.template, line 71:
https://github.com/ros2/rmw_connext/pull/100/files#diff-a2cb81bcab4593a61a1d8ad7a7308f25R71

I could change this function to take a DDSDataWriterQos struct and a DDSDataReaderQos struct, instead of an rmw_qos_profile, and initialize the DDS QoS structs in the rmw implementation before `create_requester` is called. Then the get_datareader/writer_qos functions would not need to be in the typesupport library.